tests/main/snap-session-agent-service-control/task.yaml

```diff
@@ -19,39 +19,28 @@ prepare: |
         touch agent-was-enabled
     fi
 
-    #shellcheck source=tests/lib/user.sh
-    . "$TESTSLIB/user.sh"
-    start_user_session
-    # Wait for sockets.target to finish starting so the session agent
-    # socket is available.
-    as_user systemctl --user start sockets.target
-
     # ensure curl is available (needed for e.g. core18)
     if ! command -v curl; then
         snap install --devmode --edge test-snapd-curl
         snap alias test-snapd-curl.curl curl
     fi
 
+    session-tool -u test --prepare
+    # Wait for sockets.target to finish starting so the session agent
+    # socket is available.
+    # XXX: I strongly believe this is not required anymore.
+    session-tool -u test systemctl --user start sockets.target
```

Comment:
Can you copy the XXX comment to the other tests below so they're easier to spot if we want to remove that?

Reply:
Done